idseq_dag/steps/generate_annotated_fasta.py

```diff
@@ -59,21 +76,38 @@ def get_map(m8_file):
                     sequence_name = input_fasta_f.readline()
                     sequence_data = input_fasta_f.readline()
 
-    @staticmethod
-    def generate_unidentified_fasta(input_fa, output_fa):
-        # TODO  remove annotated fasta intermediate file and replace > with : below
-        command.execute(
-            command_patterns.ShellScriptCommand(
-                script=r'''grep -A 1 '>NR::NT::' "$1" | sed '/^--$/d' > "$2";''',
-                args=[
-                    input_fa,
-                    output_fa
-                ]
-            )
-        )
-
     @staticmethod
     def old_read_name(new_read_name):
         # Inverse of new_read_name creation above.  Needed to cross-reference to original read_id
         # in order to identify all duplicate reads for this read_id.
         return new_read_name.split(":", 4)[-1]
+
+    def generate_unidentified_fasta(self, input_fa, output_fa, clusters_dict=None):
+        """
+        Generates files with all unmapped reads. If COUNT_ALL, which was added
+        in v4, then include non-unique reads extracted upstream by cdhitdup.
+        """
+        with open(output_fa, "w") as output_file:
+            for read in fasta.iterator(input_fa):
+                if not read.header.startswith(UNMAPPED_HEADER_PREFIX):
+                    continue
+
+                output_file.write(read.header + "\n")
+                output_file.write(read.sequence + "\n")
+
+                if clusters_dict:
+                    # get inner part of header like
+                    # '>NR::NT::NB501961:14:HM7TLBGX2:4:23511:18703:20079/2'
+                    line = read.header
+                    header_suffix = ""
+                    if line[-2:-1] == "/":  # /1 or /2
```

Comment:
This if is redundant with the assertions. There is no scenario where the if would reject something that would pass the assertions, you can remove 102 + 103

Reply:
sorry I don't follow. which assertions? I see how lines 103 and 105 and redundant, but the whole point of assertions is to throw an error for conditions that should be enforced elsewhere.